File: snippets/rs/array/remove clear/remove.rs
fn main()
{
    let mut array = vec![5, 4, 3, 2, 1];
    let index = 1;  // 0: first item, 1: second item...
    println!("{array:?}");  // [5, 4, 3, 2, 1]
    array.remove(index);  // ≡
    println!("{array:?}");  // [5, 3, 2, 1]

    // many items:
    let mut array = vec![5, 4, 3, 2, 1];
    let index = 1;  // 0: first item, 1: second item...
    let count = 3;
    println!("{array:?}");  // [5, 4, 3, 2, 1]
    array.drain(index..(index + count));  // ≡
    println!("{array:?}");  // [5, 1]

    // many items at the end:
    let mut array = vec![5, 4, 3, 2, 1];
    let index = 2;  // 0: first item, 1: second item...
    println!("{array:?}");  // [5, 4, 3, 2, 1]
    array.truncate(index);  // ≡
    println!("{array:?}");  // [5, 4]
}

File: snippets/php/array/remove/remove.php
<?php
$arraay = [5, 4, 3, 2, 1];
$start = 1;  // 0: first item, 1: second item...
$count = 3;
var_export($arraay);        // array (0 => 5, 1 => 4, 2 => 3, 3 => 2, 4 => 1, )
$removedItems = array_splice($arraay, $start, $count);  // ≡
var_export($arraay);        // array (0 => 5, 1 => 1, )
var_export($removedItems);  // array (0 => 4, 1 => 3, 2 => 2, )
?>